### PR TITLE
Beta svein

### PR DESCRIPTION
**Pull request by kongebra**
Partially added group functions.

Mainly pull request for this:
* Report for reviews done for assignment

### CODE REVIEW

**Review by BredeFK — CHANGES_REQUESTED** [review]
* Fix the commented issues.
* Fix `SetFlash` and `GetFlash` lint error (even though you didn't create them)
* **Bonus**: Add assignment id to report name to easier separate multiple reports.
* **Also**: Remember to add meeting notes

Other than that 👉 💯 🥇 :trollface: :neckbeard:

File: webservice/template/admin/assignment/submissions.tmpl [hunk 1]
```diff
@@ -63,6 +63,13 @@
     </div>
     {{end}}
 
+    <div class="row">
+        <div class="col">
+            <a href="/admin/assignment/{{.Assignment.ID}}/report.xlsx" class="btn btn-primary">Download Report</a>
```
[the remark this answers is not in the document]
Comment (kongebra) [reply webservice/template/admin/assignment/submissions.tmpl]: da0312e

File: webservice/template/admin/assignment/update_new.tmpl [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            });
+
+            let reviewSelect = document.getElementById('review_id');
+            let reviewDeadlineContainer = document.getElementById('review_deadline_container');
+            let reviewDeadline = document.getElementById('review_deadline');
+            let toggleReviewBtn = document.getElementById('toggleReview');
+
+            toggleReviewBtn.addEventListener('click', () => {
+                reviewSelect.selectedIndex = 0;
+                reviewDeadlineContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: don't hide the review stuff even if it is disabled. It should be enabled/disabled by the button

File: webservice/template/admin/assignment/update_new.tmpl [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                reviewDeadline.removeAttribute('required');
+                enabledInput.removeAttribute("value");
+                enabledInput.setAttribute("value", "false");
+            });
+
+            reviewSelect.addEventListener('change', e => {
+                if (e.target.value !== '0') {
+                    reviewDeadlineContainer.classList.remove('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: .

File: webservice/template/admin/assignment/update_new.tmpl [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                    reviewDeadline.setAttribute('required', '');
+                    enabledInput.removeAttribute("value");
+                    enabledInput.setAttribute("value", "true");
+                } else {
+                    reviewDeadlineContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: .

File: webservice/template/admin/assignment/update_new.tmpl [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                    enabledInput.setAttribute("value", "false");
+                }
+            });
+
+            let submissionSelect = document.getElementById('submission_id');
+            let submissionContainer = document.getElementById('submission_container');
+
+            submissionSelect.addEventListener('change', e => {
+                if (e.target.value !== '0') {
+                    submissionContainer.classList.remove('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: .

File: webservice/template/admin/assignment/update_new.tmpl [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+                } else {
+                    submissionContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: .

File: webservice/template/admin/assignment/update_new.tmpl [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+                    <select name="review_id" id="review_id" class="form-control">
+                        <option value="0">None</option>
+                        {{range $R := .Reviews}}
+                            <option value="{{$R.ID}}"{{if eq $R.ID $A.ReviewID.Int64}} selected{{end}}>{{$R.Form.Name}}</option>
+                        {{end}}
+                    </select>
+                </div>
+
+                <div id="review_deadline_container" {{if not .Assignment.ReviewID.Valid}}class="sr-only"{{end}}>
+                    <button type="button" id="toggleReview" class="btn btn-danger btn-sm mb-3">Disable</button>
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/update_new.tmpl]: Se hvordan den andre update filen er, disable/enable knappen skal være over.

**Review by JohanAanesen — CHANGES_REQUESTED** [review]
fix assignment create og assignment update sånn den var.

File: webservice/template/admin/assignment/create.tmpl [hunk 8]
```diff
@@ -189,7 +198,44 @@
...
-            deactiveReviews();
-        };
+            let reviewSelect = document.getElementById('review_id');
+            let reviewDeadlineContainer = document.getElementById('review_deadline_container');
+            let reviewDeadline = document.getElementById('review_deadline');
+            let toggleReviewBtn = document.getElementById('toggleReview');
+
+            toggleReviewBtn.addEventListener('click', () => {
+                reviewSelect.selectedIndex = 0;
+                reviewDeadlineContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/create.tmpl]: .

File: webservice/template/admin/assignment/create.tmpl [hunk 9]
```diff
@@ -189,7 +198,44 @@
...
+                reviewDeadline.removeAttribute('required');
+                enabledInput.removeAttribute("value");
+                enabledInput.setAttribute("value", "false");
+            });
+
+            reviewSelect.addEventListener('change', e => {
+                if (e.target.value !== '0') {
+                    reviewDeadlineContainer.classList.remove('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/create.tmpl]: .

File: webservice/template/admin/assignment/create.tmpl [hunk 10]
```diff
@@ -189,7 +198,44 @@
...
+                    reviewDeadline.setAttribute('required', '');
+                    enabledInput.removeAttribute("value");
+                    enabledInput.setAttribute("value", "true");
+                } else {
+                    reviewDeadlineContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/create.tmpl]: .

File: webservice/template/admin/assignment/create.tmpl [hunk 11]
```diff
@@ -189,7 +198,44 @@
...
+                    enabledInput.setAttribute("value", "false");
+                }
+            });
+
+            let submissionSelect = document.getElementById('submission_id');
+            let submissionContainer = document.getElementById('submission_container');
+
+            submissionSelect.addEventListener('change', e => {
+                if (e.target.value !== '0') {
+                    submissionContainer.classList.remove('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/create.tmpl]: .

File: webservice/template/admin/assignment/create.tmpl [hunk 12]
```diff
@@ -189,7 +198,44 @@
...
+                } else {
+                    submissionContainer.classList.add('sr-only');
```
Comment (JohanAanesen) [inline webservice/template/admin/assignment/create.tmpl]: .

File: webservice/service/assignment.go [hunk 13]
```diff
@@ -46,22 +46,32 @@ func (s *AssignmentService) FetchFromCourse(courseID int) ([]*model.Assignment,
...
 
-// HasReview func
+// HasReview checks if an assignment has review
 func (s *AssignmentService) HasReview(assignmentID int) (bool, error) {
 	assignment, err := s.assignmentRepo.Fetch(assignmentID)
 	if err != nil {
 		return false, err
 	}
 
-	return assignment.ReviewEnabled, err
```
Comment (JohanAanesen) [inline webservice/service/assignment.go]: This should use the ReviewEnabled variable, not check if reviewID is valid.

**Review by JohanAanesen — CHANGES_REQUESTED** [review]

**Review by JohanAanesen — APPROVED** [review]

**Review by BredeFK — APPROVED** [review]

**Comment by BredeFK** [comment]
[Export submission/review data to excel](https://trello.com/c/8lbbmNoh/44-export-submission-review-data-to-excel)

**Pull request merged**